File: BudgetManager(BackEnd)/src/main/java/com/BudgetManager/BudgetManager/model/Budget.java
package com.BudgetManager.BudgetManager.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@RequiredArgsConstructor

public class Budget {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    private Double limitAmount;
    private Double spentAmount;

    @OneToMany(mappedBy = "budget", cascade = CascadeType.ALL)
    private List<Transaction> transactions = new ArrayList<>();
}

File: BudgetManager/src/app/core/models/budget.ts
export interface Category {
  id?: number;
  name: string;
  limitAmount: number;
  transactions?: Transaction[];
}

export interface Transaction {
  id?: number;
  amount: number;
  date: string;
  description: string;
  category: Category;
  budget: Budget;
}

export interface Budget {
  id?: number;
  category: Category;
  limitAmount: number;
  spentAmount: number;
  transactions: Transaction[];
}
